packages/server/src/controllers/files.ts

```diff
@@ -17,8 +23,19 @@ const dummy_file = {
 
 export default class FilesController {
   static create: Middleware = async (ctx) => {
-    FilesValidator.validateForCreate(ctx)
-    await File.create('test')
+    const { body: { owner }, file } = ctx.request
+    try {
+      FilesValidator.validateForCreate(ctx)
+      for await (const ipfsFile of ipfs.add(globSource(file.path))) {
```

Comment:
Shouldn't `ipfs.add` be called only once? Or does it already support dirs upload?